pkg/server/loss_of_quorum.go

```diff
@@ -117,3 +119,48 @@ func publishPendingLossOfQuorumRecoveryEvents(
 		}
 	})
 }
+
+func maybeRunLossOfQuorumRecoveryCleanup(
+	ctx context.Context, stores *kvserver.Stores, server *Server, stopper *stop.Stopper,
+) {
+	var cleanup loqrecoverypb.DeferredRecoveryActions
+	err := stores.VisitStores(func(s *kvserver.Store) error {
+		c, found, err := loqrecovery.ConsumeCleanupActionsInfo(ctx, s.Engine())
```

Comment:
Currently it is just trying once and relying on the fact that all nodes would try to do this so one will succeed. Not a strong guarantee. Alternatively, we could keep trying for some time, maybe longer than theoretical liveness range recovery time e.g. replication queue should pick up liveness within 10 min, then it would upreplicate (which could be throttled potentially by indeterminate amount), then we'll pick it up within seconds and update.
I don't like idea of keeping this entry if we can't update liveness. It would stay there till next restart so you'll have this action silently "scheduled". It is idempotent now, so technically fine.

Reply:
Discussed offline. We'll need to get these nodes decommissioned one way or another (otherwise they could rejoin the cluster via newly added nodes or prevent later cluster upgrades), so we should keep retrying with exponential backoff and make sure the actions are idempotent. We only remove the action when it's successfully applied. In the worst case, the operator can manually perform e.g. the decommission action to complete the job.